pkg/utils/grpc.go

```diff
@@ -34,37 +59,124 @@ func NewGrpc(serverType string, port int, certPem, keyPem, rootCertPem []byte) (
 		}),
 	}
 
-	mutualTLS, err := setupMutualTLS(false, serverType, certPem, keyPem, rootCertPem)
-	if err != nil {
-		log.Error().Err(err).Msg("Error setting up mutual tls for GRPC server")
-		return nil, nil, err
+	// #nosec G402: TLS MinVersion too low
+	tlsConfig := tls.Config{
+		InsecureSkipVerify: false,
+		ServerName:         s.name,
+		ClientAuth:         tls.RequireAndVerifyClientCert,
+		MinVersion:         constants.MinTLSVersion,
+		GetConfigForClient: func(*tls.ClientHelloInfo) (*tls.Config, error) {
+			// use lock to prevent concurrent updates and reads to the tls config
+			s.mu.Lock()
+			defer s.mu.Unlock()
+			return &s.config, nil
```

Comment:
`s.config` is initially nil here; is that intentional?

Reply:
It was intentional. I want to delay issuing the ads server certificate until after we subscribe to rotations of that certificate. Therefore, we're holding off on issuing the ads cert until the caller starts the server (calls `GrpcServe`). By holding off on the certificate creation, I think it also makes sense to hold off on creating the tls config.

Reply:
Ah, I see. Just be cautious (perhaps comment) about accessing `s.config` around the caller so as to avoid nil pointer errros